### PR TITLE
Make build work better on macOS

### PR DESCRIPTION
**Pull request by mcintyre94**
- Use /tmp/lock for locks, avoiding macOS lockdown on /var/lock
- Replace sleep infinity with a loop

Misc other changes:

- Run prettier:fix in packages/library-legacy
- Fix doc directory in packages/library-legacy clean

### CODE REVIEW

File: packages/library-legacy/package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "compile:js": "cross-env NODE_ENV=production rollup -c",
     "compile:typedefs": "./scripts/typegen.sh",
     "build:fixtures": "set -ex; ./test/fixtures/noop-program/build.sh",
-    "clean": "rimraf ./docs ./declarations ./lib",
+    "clean": "rimraf ./doc ./declarations ./lib",
```
Comment (steveluscher) [inline packages/library-legacy/package.json:44]: LOL.

File: packages/library-legacy/package.json [hunk 2]
```diff
@@ -128,4 +128,4 @@
     "typedoc": "^0.23.28",
     "typescript": "^5.0.3"
   }
-}
+}
```
Comment (steveluscher) [inline packages/library-legacy/package.json:131]: giphy

File: packages/test-config/test-validator-setup.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ module.exports = async function globalSetup() {
         { command: '../../scripts/start-shared-test-validator.sh' },
         // This 'server' is a noop; we only use it to run the 'wait for server' logic.
         {
-            command: 'sleep infinity',
+            command: 'while true; do sleep 86400; done',
```
Comment (steveluscher) [inline packages/test-config/test-validator-setup.js:10]: Really Mac OS? Ugh.

File: scripts/start-shared-test-validator.sh [hunk 4]
```diff
@@ -22,7 +24,7 @@ TEST_VALIDATOR_LEDGER="$( cd "$(dirname "${BASH_SOURCE[0]}")/.." ; pwd -P )/test
       wait
     else
       echo "Sharing lock on already running test validator (PID $(pidof $TEST_VALIDATOR))"
-      sleep infinity
+      while true; do sleep 86400; done
```
Comment (steveluscher) [inline scripts/start-shared-test-validator.sh:27]: I'm concerned. The idea here is that `sleep infinity` emits a file descriptor that forms part of the lock. If `sleep 86400` finishes, does it temporarily release the lock before the loop restarts?

Yes, I know 86400 seconds is a whole day :)

**Review by steveluscher — APPROVED** [review]
Looks great! Just leaving some comments for your consideration. And that EOL.

File: scripts/start-shared-test-validator.sh (hunk 4 above)
Comment (mcintyre94) [reply scripts/start-shared-test-validator.sh:27]: Ah, got you. How about we just do like `sleep 86400000` here? I don't know what the macOS actual max is but it doesn't complain about that

**Pull request merged**